sync/bigquery_etl.py

```diff
@@ -0,0 +1,59 @@
+import collections
+import re
+from pathlib import Path
+from typing import Dict
+
+import requests
+from io import BytesIO
+import tarfile
+
+import yaml
+
+FILE_PREFIX = "bigquery-etl-generated-sql/sql"
+QUERY_FILES = {
+    "query.sql",
+    "view.sql",
+    "query.py",
+}
+METADATA_FILE = "metadata.yaml"
+SUPPORTED_PROJECTS = {
+    "moz-fx-data-shared-prod",
+    "moz-fx-data-marketing-prod",
+}
+VALID_TABLE_RE = re.compile(
+    rf"{FILE_PREFIX}\/({'|'.join(SUPPORTED_PROJECTS)})\/([a-z0-9_-]+)\/([a-z0-9_-]+)\/({'|'.join(QUERY_FILES)}|{METADATA_FILE})"
+)
+
+REPO_ARCHIVE = "https://github.com/mozilla/bigquery_etl/archive/generated-sql.tar.gz"
+REPO_URL = "https://github.com/mozilla/bigquery-etl/blob/generated-sql/sql"
+WTMO_URL = "https://workflow.telemetry.mozilla.org/dags"
+
+
+def get_bigquery_etl_table_references() -> Dict:
+    table_references = collections.defaultdict(dict)
+    # {qualified_name: {bigquery_etl_url: ..., wtmo_url: ...}}
+
+    print("Fetching metadata from GitHub...")
```

Comment:
Tiny nit (I have a slight preference for logging over print in production code 😅)
```suggestion
    logger.info("Fetching metadata from GitHub...")
```